pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -647,9 +647,19 @@ func (c *Cluster) startDeployments(mons []*monConfig, requireAllInQuorum bool) e
 		} else {
 			logger.Warningf("failed to list mon deployments. attempting to continue. %v", err)
 		}
-	} else if len(deployments.Items) < len(mons) {
+	}
+
+	readyReplicas := 0
+	// Ensuring the mon deployments should be ready
+	for _, deploy := range deployments.Items {
+		readyReplicas = int(deploy.Status.AvailableReplicas) + readyReplicas
+	}
+	if len(deployments.Items) < len(mons) {
 		logger.Infof("%d of %d expected mon deployments exist. creating new deployment(s).", len(deployments.Items), len(mons))
 		onlyCheckQuorumOnce = true
+	} else if readyReplicas == 0 {
+		logger.Infof("%d of %d expected mons are ready", readyReplicas, len(mons))
```

Comment:
Above, we report after the message `creating new deployment(s)`. Should we also report a similar message here? I think it helps to understand what the operator is intending to do next. From my understanding, this path will mean that we are `creating and/or updating deployments without checking quorum in attempt to achieve a healthy mon cluster`. Is that correct, and if so, should we report that or something similar?

Reply:
yeah updated the logging with your suggestion